datalad/config.py

```diff
@@ -85,9 +80,9 @@ def _parse_gitconfig_dump(dump, store, fileset, replace, cwd=None):
             continue
         if line.startswith('file:'):
             # origin line
-            fname = line[5:]
-            if not isabs(fname):
-                fname = opj(cwd, fname) if cwd else abspath(fname)
+            fname = Path(line[5:])
+            if not fname.is_absolute():
+                fname = Path(cwd) / fname if cwd else Path.cwd() / fname
```

Comment:
This seems incorrect to me. The dump comes from a `git config` call that is executed in `self._runner.cwd`, right? So, how would `Path.cwd()` be relevant? Shouldn't it be relative to CWD of the git-config call? This may usually be the same, but I think we tend to shoot our feet with this kind of assumption as soon as someone changes something.

Reply:
> Shouldn't it be relative to CWD of the git-config call?

That's the `if cwd` above.

Reply:
Just to clarify: Are you saying that the translation of the code to pathlib introduces a behavior change? Or are you saying what we used to do is wrong? In the latter case, I'd prefer to file that as an issue and deal with it in a dedicated PR.

Reply:
@mih: The latter.

@kyleam: Yes, but that's kind of the problem. Now we call git-config (possibly with a `cwd` argument or whatever logic) to then call a parsing helper, where we need to make sure to end up passing the same `cwd`. I think those two should be linked.

Reply:
What you just said is different than saying it's "incorrect".  You can audit the three call sites and see that they all explicitly pass `cwd`.  It's outside the scope of this PR, but I agree that tighter coupling would be better here or at the very least the probably-not-what-you-want "cwd=None => .cwd()" fallback could be dropped in favor of a mandatory argument.

Reply:
Ok, so let's keep things the way they are until a white knight comes and cleans out the cow shit ;-)